src/Decoders/DataParser.php

```diff
@@ -367,9 +367,14 @@ public function parseResource($data, $path, $resType, $loader = null)
 
             $id = ($this->hasValue($value, 'id')) ? $this->getValue($value, 'id') : null;
 
-            $pathValue = null;
+            if (null !== $id) {
+                $pathValue = $this->context->getResource($name, $id);
+                if (null !== $pathValue) {
+                    $id = null;
```

Comment:
@mixailshaulsky why we reset 'id' here? I think it will break cases when resource ID specified on client side.

Reply:
no, in this scope var $id is just used as flag to register new resource